scapy/arch/pcapdnet.py

```diff
@@ -483,21 +486,42 @@ def get_if_list():
             raise
     else:
         def get_if_raw_hwaddr(iff):
+            """Return a tuple containing the link type and the raw hardware
+               address corresponding to the interface 'iff'"""
+
             if iff == scapy.arch.LOOPBACK_NAME:
-                return (772, '\x00'*6)
+                return (ARPHDR_LOOPBACK, '\x00'*6)
+
+            # Retrieve interface information
             try:
                 l = dnet.intf().get(iff)
-                l = l["link_addr"]
+                link_addr = l["link_addr"]
             except:
-                raise Scapy_Exception("Error in attempting to get hw address for interface [%s]" % iff)
-            return l.type,l.data
+                msg = "Error in attempting to get hw address for interface"
+                msg += " [%s]" % iff
```

Comment:
You should avoid unneeded string concatenation, optionally using a trailing `\`:
```Python
                msg = "Error in attempting to get hw address for interface" \
                      " [%s]" % iff
                raise Scapy_Exception(msg)
```

Or even better (no temporary variable, no trailing backslash):
```Python
                raise Scapy_Exception("Error in attempting to get hw address"
                                      " for interface [%s]" % iff)
```